### PR TITLE
611 Simplify Hiero tag workflow

### PR DESCRIPTION
**Pull request by jezscha**
- [x] refactoring Tag workflow - removing json, adding pype tag for single taging
- [x] adding functions to lib for plugin wide use
- [x] Creator with gui and renamer for hierarchical data use
- [x] Publishing plugins refactoring for single tag collection
- [x] Collectors in publishing refactoring for pre-collecting faze of publisher
- [x] Loading clip to timeline with update and remove, containerized and activated Manage menu item

## What to test in Shot Creator plugin ui?
- Renaming with Hiero tool and only generating tag in one layer > select clips in single layer which are renamed with hiero's build in renaming tool (select clips / Editorial / Rename) and check if the P. Tag was created with correct asset. It should be the same as the original name of clip after rename > Rename clips False
- Renaming with tool and layer names defining subset names with vertical synchronization > create clips in multiple layers. Rename layers to subset name. Slips should be exactly above each other. Select all clips in layers. In UI panel set > Vertical rename True, Master track (Usually `main`), Subset name <track_name>
- Renaming with tool and no vertical synchronization used (if all clips across layers are individual assets) > Vertical rename False, Subset name `main`

### CODE REVIEW

**Comment by mkolar** [comment]
I would simplify a few bits even more in the GUI the amount of categories gets somewhat confusing. How about merging a few to get this for the hierarchy and rename section:
image

For the shot attributes I'd rename this way and add source resolution option to get rid of another tag
image

**Review by mkolar — CHANGES_REQUESTED** [review]
I tried to do the changes myself, but couldn't force the approach to order the items for me correctly. Maybe I've done something wrong, but it looks like the order attribute in the dictionary is not really affecting the order. 

I'm considering this a prototype GUI, so let's please just fix the basics and move on with the functionality.

File: pype/plugins/hiero/create/create_shot_clip.py [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+                "subsetName": {"value": ["main", "bg", "fg", "bg",
+                                         "animatic", "<track_name>"],
+                               "type": "QComboBox",
+                               "label": "Subset Name", "target": "ui", "toolTip": "chose subset name patern, if <track_name> is selected, name of track layer will be used",  # noqa
+                                "order": 0},
+                "subsetFamily": {"value": ["plate", "take"],
+                                 "type": "QComboBox",
+                                 "label": "Subset Family", "target": "ui", "toolTip": "What use of this subset is for",  # noqa
+                                  "order": 1},
+                "previewOn": {"value": True, "type": "QCheckBox",
```
[the remark this answers is not in the document]
Comment (jezscha) [reply pype/plugins/hiero/create/create_shot_clip.py]: I am not sure if it is not rather preview mov files rather then review workflow. This is for ftrack preview video clip.

**Comment by jezscha** [comment]
> I would simplify a few bits even more in the GUI the amount of categories gets somewhat confusing. How about merging a few to get this for the hierarchy and rename section:
These sections were kept serparatelly as they are two different thing. Usually there could be case where renaming is not needed and we would rather use the build in renaming. Although the hierarchy will be used in all cases.

**Comment by mkolar** [comment]
I can't get the renamer to work at all. The tags are created and the data seem to reflect everything correctly.

**Comment by mkolar** [comment]
Collecting goes through correctly now. Metadata is correct.

When renaming using the tool, it only renames the first 2 clips correctly and all following then get the same name as clip number 2.

**Comment by mkolar** [comment]
Very nice. Much much easier to use and set up shot. I've made a small tweak to default values, and this is ready for merging of the subsequent PRs into here. 
My changes:
removed episode from default
shortened the shot padding
changes sequence default to `sq01` from sc020` 
image

**Review by mkolar — APPROVED** [review]

**Comment by mkolar** [comment]
this pr was cleaned up and replicated in #907 so we can close this

**Pull request closed without merging**